File: Kladd/timer.c
#include <time.h>
#include <assert.h>

static time_t g_startTime = -1;

void
timer_start(){
    g_startTime = time(0);
    assert(g_startTime != -1);
}


void 
timer_stop(){
  g_startTime = -1;
}

int 
timer_isTimeOut(){
  if(g_startTime <0){
    // There is no timeout, because the timer is not started
    return 0;
  }

  time_t now = time(0);
  if(now - g_startTime > 3){
    return 1;
  }else{
    return 0;
  }
}

File: Kladd/timer.go
package timer

var timerFlag int
var t time.Time

func timer(chan_timer chan bool){
	for{
		if timerFlag = 1 {
			t = time.now()
			timerFlag = 0
		}

		if time.now() - t > 3{
			chan_timer <- true
		}
	}
}

func start() {
	timerFlag = 1
}